extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/jsonobject/JsonObjectValueResolver.java

```diff
@@ -48,7 +48,7 @@ public CompletionStage<Object> resolve(EvalContext context) {
             default:
                 return jsonObject.containsKey(context.getName())
                         ? CompletableFuture.completedFuture(jsonObject.getValue(context.getName()))
-                        : Results.notFound(context);
+                        : Results.NULL;
```

Comment:
Hm, I think that `Results.notFound(context)` is more correct in this case. We use the same strategy for `Map` - if it does not contain a key of the given name the `NotFound` is returned.

In general, the difference between `null` and `NotFound` is that `null` is a _valid value_ whereas `NotFound` usually means [missing property](https://quarkus.io/guides/qute-reference#missing-properties).

Now, if [Strict Rendering](https://quarkus.io/guides/qute-reference#strict_rendering) is enabled (default) then  a missing property will always result in a `TemplateException`.  However, you can use _default values_ and _safe expressions_ in order to suppress the error, e.g. `{#if foo.bar??}` which translates to `{#if foo.bar.or(null)}` -> if `foo` is a `JsonObject` that does not contain the property "bar" then `null` is used and since `null` is considered `falsy` the content of the the `{#if}` section is not rendered.